### ISSUE TITLE
Optimize VLM PatchEmbed

### ISSUE THREAD

**Issue by dayo09** [Samsung/TICO#475]
## What?

Let's optimize the below VLM PatchEmbed module.

### Q. Doesn't one-optimize resolve this?
- I checked that both 

- [x] (1) Remove redundant SelectV2 https://github.com/Samsung/TICO/pull/474
  - Remove Reshape + Concat + Reshape at the bottom of the graph too
- [ ] (2) Lower 4-dim permute to 3-dim permute

**Comment by llFreetimell** [comment]
## Idea with using Conv2D

If `kernel_size == (D, H, W)`, `input * kernel` is exactly same with element-wise multiplication of flattened elements.
Although we can optimize with `Mul` operator, let's use `Conv2D` :)

### PyTorch Codes

```python
import torch

# Original Input
N = 123
input = torch.randint(0, 3, (N, 1536)).float()

# Original Qwen3VLVisionPatchEmbed.forward()
I1 = input.view(-1, 3, 2, 16, 16)
conv3d = torch.nn.Conv3d(in_channels=3, out_channels=1024, kernel_size=(2, 16, 16), stride=(2, 16, 16), padding=0)
R1 = conv3d(I1).view(-1, 1024)

# Suggestion : Reshape + Conv2D + Permute + Unsqueeze
I2=input.view(1, 1, -1, 1536)
conv2d = torch.nn.Conv2d(in_channels=1, out_channels=1024, kernel_size=(1, 1536))
conv2d.weight.data = conv3d.weight.data.view(1024, 1, 1, 1536) # constant folding / no reshape
conv2d.bias.data = conv3d.bias.data
conv2d_result = conv2d(I2)
R2 = torch.permute(conv2d_result, (2, 1, 0, 3)).view(-1, 1024)

# You can check R1 == R2
# print(R1)
# print(R2)
```

### Needed passes

- Add above special pattern to `ConvertConv3dToConv2d` pass
- Implement `ConvertPermuteToReshape` pass
  - e.g. when `x.shape == [1, a, 1, b]`, `torch.permute(x, [1,3,0,2])` can be changed to `torch.reshape(x, [a, b, 1, 1])`
